### ISSUE TITLE
Add the option to automatically add played songs in the library

### ISSUE THREAD

**Issue by ghost** [z-huang/music#1057]
### Checklist

- [X] I've checked that there is no other issue about this feature request.
- [X] This issue contains only one feature request.
- [X] The title of this issue accurately describes the feature request.

### Feature description

<details>
  <summary>Previous Request. Disregard this and read the request below instead.</summary>

# Ditch the "Library"

In my opinion, the way the app handles the "Library" is just cumbersome. I propose the following changes to streamline and optimize how the application handles the user's media.

#  Songs Tab:

Instead of having to manually add a song one-by-one to the library, just add every song the user interacts with. By default, display every song the user has played, liked, or downloaded. After that, the user can just apply filters/sort as needed. Do not display songs the user haven't interacted with, even if they follow the artist.

### Filters & Sorting

**The chips can be simultaneously toggled.**

- Sort (Drop-down) - same as before, but add "Last Played" 
- Liked (Chip) - songs the user liked
- Downloaded (Chip) - songs that have been cached
- Followed-only (Chip) - if toggled on, **do not** display the song if the user doesn't follow the artist

# Artists Tab:

For the artists, introduce a "Follow" button alongside "Liked". Follow works the same as a subscribe button, while "Like" (Heart) treats the artist as a favorite. **A liked artist is automatically followed.**

By default, display every artist the user follows, then apply filters/sort as needed.

### Filters & Sorting

- Sort (Drop-down) - sort the artists based on their name, date added, etc.
- Liked (Chip) - shows the favorite artists of a user 

# Albums Tab

For the Albums tab, introduce a "Follow" or "Save" button alongside "Like". Following an album is essentially just saving it in a "Library", but liking marks it is a favorite. **A liked album is automatically followed.**

By default display every album the user follows, then apply filters/sort as needed.

### Filters & Sorting

- Sort (Drop-down) - sort the albums based on release, name, date added, etc.
- Liked (Chip) - display only the favorite albums of the user

</details>

Giving it more thought, considering that the previous request will introduce breaking changes, I have changed my feature request to:

- Introduce an "All" chip for the songs tabs. The "All" chip will show every song the user has interacted with (played, added to library, liked, or downloaded). 
- Or, add an option to automatically add played songs to the Library.

I have placed the previous request in a spoiler and edited the title to reflect the changes.

### Why do you want this feature?

I believe that the current way the app handles the library is too cumbersome. The user has to manually add each and every song to the library.

### Additional information

None

**Comment by v2302** [comment]
Some additions:

- "Downloaded" chips can also be added to the Artists and Albums tab to filter Artists/Albums with downloads.
- Remvoving the library would introduce a conflict with the previous feature of being able to automatically download a song once added to the library, as such, a replacement would be to offer a "Download/Cache song upon playback" toggle in the settings.

**Comment by z-huang** [comment]
The library serves as a repository for your preferred songs. To access songs that you have previously played, you may navigate to the history screen, which should adequately fulfill your requirements. If you have lots of songs to add to the library, swipe action (#963) should be what you're looking for.

**Comment by ghost** [comment]
> The library serves as a repository for your preferred songs. To access songs that you have previously played, you may navigate to the history screen, which should adequately fulfill your requirements. If you have lots of songs to add to the library, swipe action (#963) should be what you're looking for.

That is true, I guess a more appropriate explanation for my request would be that I want to have a centralized view for all my songs (Played, Library, Liked, Downloaded), and be able to sort/filter them, because currently, they are all separate (aside from liked songs being automatically added to the library).

#963 would be great, especially if it can be customizable (e.g., add to queue, add to playlist, add to library, like, download, etc.). That along with #695 should make batch operations easier.
